bim2sim/task/bps/sb_creation.py

```diff
@@ -14,33 +14,38 @@
 from bim2sim.kernel.finder import TemplateFinder
 from bim2sim.kernel.units import ureg
 from bim2sim.task.base import ITask
-from bim2sim.workflow import Workflow
+from bim2sim.simulation_settings import BaseSimSettings
 
 logger = logging.getLogger(__name__)
 
 
 class CreateSpaceBoundaries(ITask):
     """Create space boundary elements from ifc."""
 
-    reads = ('ifc', 'instances', 'finder')
+    reads = ('ifc_files', 'instances')
     touches = ('space_boundaries',)
 
-    def run(self, workflow, ifc, instances, finder):
+    def run(self, ifc_files, instances):
         logger.info("Creates elements for IfcRelSpaceBoundarys")
         type_filter = TypeFilter(('IfcRelSpaceBoundary',))
-        entity_type_dict, unknown_entities = type_filter.run(ifc)
-        instance_lst = self.instantiate_space_boundaries(
-            entity_type_dict, instances, finder,
-            workflow.create_external_elements, workflow.ifc_units)
-        bound_instances = self.get_parents_and_children(workflow, instance_lst,
-                                                        instances)
-        instance_lst = list(bound_instances.values())
-        logger.info("Created %d elements", len(bound_instances))
-
-        space_boundaries = {inst.guid: inst for inst in instance_lst}
+        space_boundaries = {}
+        for ifc_cls in ifc_files:
+            entity_type_dict, unknown_entities = type_filter.run(ifc_cls.file)
+            instance_lst = self.instantiate_space_boundaries(
+                entity_type_dict, instances, ifc_cls.finder,
+                self.playground.sim_settings.create_external_elements,
+                ifc_cls.ifc_units)
+            bound_instances = self.get_parents_and_children(
+                self.playground.sim_settings, instance_lst, instances)
+            instance_lst = list(bound_instances.values())
+            logger.info(f"Created {len(bound_instances)} bim2sim SpaceBoundary "
+                        f"instances based on IFC file: {ifc_cls.ifc_file_name}")
+            space_boundaries.update({inst.guid: inst for inst in instance_lst})
+        logger.info(f"Created {len(space_boundaries)} bim2sim SpaceBoundary "
+                    f"instances in total for all IFC files.")
```

Comment:
I would rather not name it ifc_cls as it sounds like these are individual classes within the ifc rather then different ifc_files.
```suggestion
        space_boundaries = {}
        for ifc_file in ifc_files:
            entity_type_dict, unknown_entities = type_filter.run(ifc_file.file)
            instance_lst = self.instantiate_space_boundaries(
                entity_type_dict, instances, ifc_file.finder,
                self.playground.sim_settings.create_external_elements,
                ifc_file.ifc_units)
            bound_instances = self.get_parents_and_children(
                self.playground.sim_settings, instance_lst, instances)
            instance_lst = list(bound_instances.values())
            logger.info(f"Created {len(bound_instances)} bim2sim SpaceBoundary "
                        f"instances based on IFC file: {ifc_file.ifc_file_name}")
            space_boundaries.update({inst.guid: inst for inst in instance_lst})
        logger.info(f"Created {len(space_boundaries)} bim2sim SpaceBoundary "
                    f"instances in total for all IFC files.")
```

Reply:
solved with 12e7c1e157452167063606fffc3874e9d2b75c68 as this was also done in `common_tasks.py`